code/cert/cert_pkcs.c

```diff
@@ -651,7 +663,8 @@ void pkcs_lookup_token_cert(LPCSTR szCert, CK_SESSION_HANDLE_PTR phSession, CK_O
 		}
 
 		// cleanup
-		pFunctionList->C_CloseSession(pSlotList[iSlot]);
+		*phSession = hSession;
+		//pFunctionList->C_CloseSession(pSlotList[iSlot]);
```

Comment:
Can you explain both these lines?  If a certificate was found, it would have returned before hitting this point.

Reply:
I commit this modification by error, sorry.
Please ignore this change.